File: Logic/Snippets/Java/Interface_callback.java
// Interfaceを利用したコールバック

interface CallBackListener {
    public void onFinishedHoge(boolean isSuccess, String errorCode);
}

class User {
    void execute(CallBackListener callback) {
        System.out.println("execute");
        if (callback != null) {
            callback.onFinishedHoge(true, "NO_ERROR");
        }
    }
}

public class Interface_callback {

    public static void main(String args[]) {
        
        User user = new User();
        user.execute(new CallBackListener() {
            // コールバック処理
            @Override
            public void onFinishedHoge(boolean isSuccess, String errorCode) {
                System.out.println("onFinishedHoge!!!");
            }
        });
    }
}

File: Logic/Snippets/Kotlin/Interface_callback.kt
// Interfaceを利用したコールバック

interface CallBackListener {
    fun onFinishedHoge(isSuccess: Boolean, errorCode: String)
}

class User() {
    fun execute(callback: CallBackListener) {
        println("execute")
        callback.onFinishedHoge(true, "NO_ERROR")
    }
}

fun main() {
    val user = User()
    user.execute(object : CallBackListener {
        override fun onFinishedHoge(isSuccess: Boolean, errorCode: String) {
            println("complete!!! ${isSuccess}")
        }
    }) 
}